app/controllers/api/v1/courses_controller.rb

```diff
@@ -0,0 +1,120 @@
+include ActionController::Flash
+
+class Api::V1::CoursesController < ApplicationController
+  before_action :set_course, only: %i[ show update destroy ]
+
+  # GET /courses
+  # List all the courses
+  def index
+    @courses = Course.all
+
+    render json: @courses
+  end
+
+  # GET /courses/1
+  # Get a course
+  def show
+    render json: @course
+  end
+
+  # POST /courses
+  # Create a course
+  def create
+    @course = Course.new(course_params)
+
+    if @course.save
+      render json: @course, status: :created
+    else
+      render json: @course.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /courses/1
+  # Update a course
+  def update
+    if @course.update(course_params)
+      render json: @course
+    else
+      render json: @course.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /courses/1
+  # Delete a course
+  def destroy
+    @course.destroy
+    render json: { message: "Course with id #{params[:id]}, deleted" }, status: :ok
+  end
+
+  # Adds a Teaching Assistant to the course
+  def add_ta
+    @course = Course.find(params[:id])
+    @user = User.find_by(id: params[:ta_id])
+    if @user.nil?
+      return render json: { status: "error", message: "The user " + params[:ta_id].to_s + " does not exist" }, status: :bad_request
+    elsif !TaMapping.where(ta_id: @user.id, course_id: @course.id).empty?
+      return render json: { status: "error", message: "The user " + params[:ta_id].to_s + " is already a TA for this course." }, status: :bad_request
+    else
+      @ta_mapping = TaMapping.create(ta_id: @user.id, course_id: @course.id)
+      @role_id = Role.find_by(name: 'Teaching Assistant').id
+      @user.update_attribute(:role_id, @role_id)
+      @user.save
+    end
+
+    if @ta_mapping.save
+      render json: @ta_mapping.slice(:course_id, :ta_id), status: :created
+    else
+      render json: @ta_mapping.errors, status: :unprocessable_entity
+    end
+  end
+
+  # Displays all Teaching Assistants for the course
+  def view_tas
+    @course = Course.find(params[:id])
```

Comment:
To streamline the code, consider adding `view_ta` to the list of actions in the `before_action: set_course` method instead of creating a new local variable for the course.

Reply:
Yes, line 73 can be removed by adding view_tas to the list of actions.